opentelemetry/src/common.rs

```diff
@@ -462,6 +462,28 @@ impl InstrumentationScope {
             attributes: None,
         }
     }
+
+    /// Instrumentation library name.
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    /// Instrumentation library version.
+    pub fn version(&self) -> Option<&str> {
+        self.version.as_ref().map(AsRef::as_ref)
+    }
+
+    /// [Schema URL] used by this library.
+    ///
+    /// [Schema URL]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/schemas/overview.md#schema-url
+    pub fn schema_url(&self) -> Option<&str> {
+        self.schema_url.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Instrumentation scope attributes to associate with emitted telemetry.
+    pub fn attributes(&self) -> &[KeyValue] {
+        &self.attributes
+    }
```

Comment:
Should we consider returning an iterator instead? This would give us flexibility to modify the internal storage later, for example, switching to non-contiguous storage or using something like [GrowableArray](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/growable_array.rs).